In [0]:

import pandas as pd
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_absolute_error
from sklearn.metrics import accuracy_score
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_absolute_error
from sklearn.metrics import accuracy_score
from sklearn.grid_search import GridSearchCV
from sklearn import datasets, svm
import matplotlib.pyplot as plt
from sklearn.model_selection import LeaveOneOut
#df = pd.read_csv('wml_data.csv', header = 0)
#df = pd.read_csv('wml_avn_fan_op.csv', header = 0)

C:\Users\Public\lib\site-packages\sklearn\cross_validation.py:44: DeprecationWarning: This module was deprecated in version 0.18 in favor of the model_selection module into which all the refactored classes and functions are moved. Also note that the interface of the new CV iterators are different from that of this module. This module will be removed in 0.20.
  "This module will be removed in 0.20.", DeprecationWarning)
C:\Users\Public\lib\site-packages\sklearn\grid_search.py:43: DeprecationWarning: This module was deprecated in version 0.18 in favor of the model_selection module into which all the refactored classes and functions are moved. This module will be removed in 0.20.
  DeprecationWarning)


In [0]:
#multivariate linear regression
#80/20 split- 20% training data
'''

array = df["WML-BR2"].tolist()

array
array_grouped=[]
for idx, item in enumerate(array):
    if  (item>0) & (item<1424) :
        array_grouped.append(1) ;
    elif (item>=1424) & (item<2849):
        array_grouped.append(2) 
    elif item>2849:
        array_grouped.append(3) ;#print(item)
print(array_grouped,array)
df['wml'] = array_grouped
#train, test = train_test_split(fifa_dataset, test_size=0.2)  // fifa_dataset here is a panda dataframe

train, test = train_test_split(df, test_size=0.1)
df.tail(2)

SyntaxError: EOF while scanning triple-quoted string literal (<ipython-input-27-d35404fcef0c>, line 21)

In [0]:
#independent and dependent variables
##features= ['Agility', 'Acceleration', 'Balance','Reactions','Positioning','Skill Moves','BallControl','Crossing','Finishing']
##target = 'SprintSpeed'
#target
'''
data_avn=train.iloc[:,[12,13,14,15,16,17,30,31,32,33,34,35]]
data_avn_fan = train.iloc[:,[12,13,14,15,16,17,30,31,32,33,34,35,18,19,20,21,22,23, 36,37,38,39,40,41  ]]
data_fan=train.iloc[:,[18,19,20,21,22,23, 36,37,38,39,40,41]]

target_1col = train.iloc[:,[1]]
#target_1col = train['wml'] 
#print(data_avn_fan, target_1col)

In [0]:
#multivariate linear regression NEW BLOCK 
#80/20 split- 20% training data

#train, test = train_test_split(fifa_dataset, test_size=0.2)  // fifa_dataset here is a panda dataframe



data_avn = pd.read_csv('wml_avn_label.csv', header = 0)

data_fan = pd.read_csv('wml_fan_label.csv', header = 0)

data_avn_fan = pd.read_csv('wml_avn_fan_label.csv', header = 0)

data_avn_fan_op = pd.read_csv('wml_avn_fan_op_label.csv', header = 0)

data_op = pd.read_csv('wml_op_label.csv', header = 0)
#label = pd.read_csv('wml_label.csv', header = 0)





# for avn_fan_op
df2 = data_avn_fan_op  ##############################################################################

train, test = train_test_split(df2, test_size=0.1)

y_train_true = train.iloc[:,[0]]

y_test_true = test.iloc[:,[0]]

x_train = train.iloc[:,1:22] #-----------------------------------------------

x_test = test.iloc[:,1: 22]

print( x_train , y_train_true)



     0       0.1  0.2      0.3       0.4  0.5      0.6   1  0.7  1.1  ...   \
76   2  0.333330    0  0.77850  1.083300  0.0  1.78160   3    4    1  ...    
70   1  0.083333    0  0.28868  0.666670  0.0  1.23090   4    4    5  ...    
87   2  0.166670    0  0.57735  0.416670  0.0  0.79296   3    2    8  ...    
51   0  0.000000    0  0.00000  0.166670  0.0  0.57735   1    2    9  ...    
25   3  0.333330    0  0.88763  0.666670  0.0  0.98473   4    2    3  ...    
2    0  0.000000    0  0.00000  0.000000  0.0  0.00000   1    0    1  ...    
68   1  0.083333    0  0.28868  0.083333  0.0  0.28868  12    1    8  ...    
39   0  0.000000    0  0.00000  0.000000  0.0  0.00000   1    0    1  ...    
30   4  0.583330    0  1.24010  0.333330  0.0  1.15470   8    4    9  ...    
110  2  0.333330    0  0.65134  0.166670  0.0  0.57735   4    2    4  ...    
17   2  0.416670    0  0.79296  0.000000  0.0  0.00000   5    0    1  ...    
120  1  0.083333    0  0.28868  0.083333  0.0  0.28868   6    1 

In [0]:
# for avn_fan
df2 = data_avn_fan  ##############################################################################

train, test = train_test_split(df2, test_size=0.1)

y_train_true = train.iloc[:,[0]]

y_test_true = test.iloc[:,[0]]

x_train = train.iloc[:,1:17] #-----------------------------------------------

x_test = test.iloc[:,1: 17]

print( x_train , y_train_true)

     0       0.1  0.2      0.3       0.4  0.5      0.6   1  0.7  1.1  0.8  \
18   2  0.250000    0  0.62158  0.166670  0.0  0.38925   9    1    4    0   
22   1  0.083333    0  0.28868  0.166670  0.0  0.57735   4    2    9    1   
71   0  0.000000    0  0.00000  0.416670  0.0  0.79296   1    2    1    0   
76   2  0.333330    0  0.77850  1.083300  0.0  1.78160   3    4    1    4   
44   1  0.083333    0  0.28868  0.000000  0.0  0.00000   3    0    1    1   
77   2  0.166670    0  0.57735  1.666700  2.0  1.66970  10    4    3    0   
21   1  0.083333    0  0.28868  0.166670  0.0  0.57735   3    2    3    1   
95   0  0.000000    0  0.00000  0.500000  0.0  0.79772   1    2    6    0   
38   2  0.250000    0  0.62158  0.416670  0.0  0.79296   4    2    4    3   
78   1  0.083333    0  0.28868  0.916670  0.0  1.56430   5    4    3    1   
29   0  0.000000    0  0.00000  0.000000  0.0  0.00000   1    0    1    0   
53   0  0.000000    0  0.00000  0.166670  0.0  0.57735   1    2    9    0   

In [0]:
# for avn
df2 = data_avn  ##############################################################################

train, test = train_test_split(df2, test_size=0.1)

y_train_true = train.iloc[:,[0]]

y_test_true = test.iloc[:,[0]]

x_train = train.iloc[:,1:8] #-----------------------------------------------

x_test = test.iloc[:,1: 8]

print( x_train , y_train_true)

     0   1  0.1  0.2
127  0   1    0    0
93   2   4    2    0
124  0   1    0    0
0    0   1    0    0
73   4   4    7    0
57   2   1    4    4
1    0   1    0    0
37   0   1    0    0
96   1   4    1    0
90   4   4    5    0
125  0   1    0    0
83   3   4    3    8
19   4   8    0   12
65   2   8    2    2
101  0   1    0    0
50   2   8    0    2
106  0   1    0    0
69   0   1    0    0
17   2   5    3    2
97   3   4    5    0
100  2  10    0    2
112  0   1    0    0
78   1   5    1    0
109  1   3    2    0
80   2   4    4    5
16   0   1    0    0
9    2   9    0    2
79   2   3    5    0
31   0   1    0    0
4    0   1    0    0
..  ..  ..  ...  ...
8    1   5    1    0
95   0   1    0    0
21   1   3    1    0
39   0   1    0    0
77   2  10    0    2
89   4   9    2    8
122  1   6    1    0
42   0   1    0    0
32   1   3    2    0
74   4   9    4   12
28   1   1    1    0
20   1   8    0    2
35   2   2    3    0
82   2   3    2    0
75   2   3    6    0
115  1   4   

In [0]:
# for fan
df2 = data_fan  ##############################################################################

train, test = train_test_split(df2, test_size=0.1)

y_train_true = train.iloc[:,[0]]

y_test_true = test.iloc[:,[0]]

x_train = train.iloc[:,1:8] #-----------------------------------------------

x_test = test.iloc[:,1: 8]

print( x_train , y_train_true)

In [0]:
# for op
df2 = data_op  ##############################################################################

train, test = train_test_split(df2, test_size=0.1)

y_train_true = train.iloc[:,[0]]

y_test_true = test.iloc[:,[0]]

x_train = train.iloc[:,1:5] #-----------------------------------------------

x_test = test.iloc[:,1: 5]

print( x_train , y_train_true)

     0   1  0.1  0.2
22   0   1    0    0
5    1   5    1    0
98   1  10    0    1
12   0   1    0    0
117  2   7    1    2
37   2   1    2    4
90   2   5    2    0
26   2   8    0    2
69   2   8    0    2
11   0   1    0    0
120  0   1    0    0
49   2   1    2    0
122  0   1    0    0
111  0   1    0    0
109  0   1    0    0
126  0   1    0    0
51   1   8    0    1
78   4   2    8    4
9    1   3    1    0
67   2   2    3    0
40   4   8    4    6
94   1   8    0    1
101  2   8    0    4
21   2   2    2    0
46   0   1    0    0
24   4  10    1    4
60   0   1    0    0
13   0   1    0    0
131  0   1    0    0
119  2   7    0    4
..  ..  ..  ...  ...
81   2   3    2    2
33   2   7    0    3
4    0   1    0    0
100  0   1    0    0
87   0   1    0    0
47   0   1    0    0
23   0   1    0    0
110  2   3    2    0
6    0   1    0    0
105  1   9    0    1
74   2   5    2    4
79   4   1    4    0
43   0   1    0    0
116  1   9    0    1
121  2   8    1    2
18   1   3   

In [0]:
from sklearn import svm
from sklearn.model_selection import LeaveOneOut 
#Create a svm Classifier
#model = svm.SVC(kernel='linear',cv=LeaveOneOut(), verbosity=2, n_jobs=8) # Linear Kernel
model = svm.SVC(kernel='linear')
##tpot = TPOTClassifier(generations=100, population_size=100, cv=LeaveOneOut(), verbosity=2, n_jobs=8)



In [0]:
'''
from sklearn.model_selection import LeaveOneOut 
loo = LeaveOneOut()
ytests = []
ypreds = []

for train_idx, test_idx in loo.split(data_avn_fan):
    X_train, X_test = X_array[train_idx], X_array[test_idx] #requires arrays
    y_train, y_test = y_array[train_idx], y_array[test_idx]
    
    model = LinearRegression()
    model.fit(X = X_train, y = y_train) 
    y_pred = model.predict(X_test)
        
    # there is only one y-test and y-pred per iteration over the loo.split, 
    # so to get a proper graph, we append them to respective lists.
        
    ytests += list(y_test)
    ypreds += list(y_pred)
        
rr = metrics.r2_score(ytests, ypreds)
ms_error = metrics.mean_squared_error(ytests, ypreds)
        
print("Leave One Out Cross Validation")
print("R^2: {:.5f}%, MSE: {:.5f}".format(rr*100, ms_error))
'''

# GRID SEARCH hyperparameter tuning

In [0]:
'''
from sklearn import svm
from sklearn.model_selection import LeaveOneOut 
import numpy as np
from sklearn.grid_search import GridSearchCV
from sklearn import datasets, svm
import matplotlib.pyplot as plt

from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_absolute_error

from sklearn.metrics import accuracy_score

parameter_candidates = [
  {'C': [1, 10, 100, 1000], 'kernel': ['linear']},
  {'C': [1, 10, 100, 1000], 'gamma': [0.001, 0.0001], 'kernel': ['rbf']},
]
# Create a classifier object with the classifier and parameter candidates
model = GridSearchCV(estimator=svm.SVR(), param_grid=parameter_candidates, n_jobs=-1)

model.fit(data_avn_fan, target_1col)
#mean absolute value for training data
##data = train[target]
true_target = target_1col
##predict =  model.predict(train[features])
predict =  model.predict(data_avn_fan)



'''


ImportError: cannot import name 'mean_square_error'

# GENERAL MODEL fit & Training cell

In [0]:
'''
x_train =  #-----------------------------------------------

x_test = 

y_train_true = 
y_test_pred
y_test_true =  #----------------------------------------
'''
model.fit(x_train, y_train_true.values.ravel()) ###########################################

#mean absolute value for training data
##data = train[target]

 
 
##predict =  model.predict(train[features])
y_train_pred =  model.predict(x_train)  #--------------------------------------------------



print (y_train_true.values.ravel(),y_train_pred)

y_test_pred = model.predict(x_test)

print(y_test_true.values.ravel(), y_test_pred)

[2 2 1 2 1 2 1 3 2 3 3 1 2 2 1 3 1 1 1 1 3 1 3 3 1 3 3 1 3 3 2 3 3 3 3 2 1
 2 3 1 3 3 3 1 3 3 2 3 1 1 3 2 3 3 3 1 2 1 1 3 2 2 3 2 3 2 3 2 1 2 3 3 2 2
 1 2 3 2 3 1 3 1 2 1 3 3 1 1 2 2 3 1 1 1 3 2 2 1 1 1 3 2 1 1 1 2 3 3 1 1 2
 3 2 1 1 2 1 1] [3 2 3 2 1 2 3 3 2 3 3 2 1 3 1 1 1 3 1 1 1 1 3 3 1 3 3 3 1 3 1 1 3 3 3 1 1
 1 3 1 3 3 3 1 1 3 3 3 1 1 1 1 2 3 3 1 1 1 1 3 2 2 3 2 3 3 3 1 1 2 1 2 1 1
 1 3 3 1 3 2 3 3 3 1 2 3 1 1 1 1 3 1 1 1 3 1 3 3 2 2 3 3 1 1 3 3 3 3 1 1 1
 2 2 1 1 3 1 1]
[1 1 3 2 2 2 1 1 1 1 3 1 2 3] [3 2 2 1 2 3 2 1 1 2 3 1 1 1]


In [0]:
accuracy = r2_score(y_test_true, y_test_pred)

NameError: name 'r2_score' is not defined

# accuracy cell 

In [0]:
from sklearn import metrics
print("Accuracy:",metrics.accuracy_score(y_test_true, y_test_pred))

print("Train Accuracy:",metrics.accuracy_score(y_train_true, y_train_pred))


from sklearn.metrics import confusion_matrix 
from sklearn.metrics import accuracy_score 
from sklearn.metrics import classification_report 

results = confusion_matrix(y_test_true, y_test_pred)
print ('Confusion Matrix :')
print(results) 
print ('Accuracy Score :',accuracy_score(y_test_true, y_test_pred) ) 
print ('Report : ')
print (classification_report(y_test_true, y_test_pred) )

Accuracy: 0.357142857143
Train Accuracy: 0.610169491525
Confusion Matrix :
[[3 3 1]
 [2 1 1]
 [1 1 1]]
Accuracy Score : 0.357142857143
Report : 
             precision    recall  f1-score   support

          1       0.50      0.43      0.46         7
          2       0.20      0.25      0.22         4
          3       0.33      0.33      0.33         3

avg / total       0.38      0.36      0.37        14



# K-fold cross validation training & score

In [0]:
from sklearn.model_selection import cross_val_score
#scores = cross_val_score(model , X = x_train , y = y_train_true , cv = x_train.shape[0]) 
scores = cross_val_score(model , X = x_train , y = y_train_true.values.ravel() , cv = 30) 
print (scores)

[ 0.33333333  0.          0.5         0.6         0.4         0.4         0.8
  0.6         0.6         0.4         0.8         0.2         1.          0.
  0.33333333  1.          0.33333333  0.          0.          0.33333333
  0.33333333  0.66666667  0.33333333  0.33333333  0.66666667  0.66666667
  0.33333333  0.33333333  1.          0.33333333]


# LEAVE ONE OUT USING SVM (named as model) 100%  data ONLY AVN & FAN

In [0]:
parameter_candidates = [
  {'C': [1, 10, 100, 1000], 'kernel': ['linear']},
  {'C': [1, 10, 100, 1000], 'gamma': [0.001, 0.0001], 'kernel': ['rbf']},
]
# Create a classifier object with the classifier and parameter candidates
model = GridSearchCV(estimator=svm.SVC(), param_grid=parameter_candidates, n_jobs=-1)



from sklearn.model_selection import LeaveOneOut ########LEAVE ONE OUT
import numpy as np
#x = np.array([[1, 2], [3, 4]])
x_train = data_avn_fan.iloc[:,1:17] #----------------
y_train = data_avn_fan.iloc[:,[0]]


x = np.array(x_train)
y = np.array(y_train.values.ravel())
loo = LeaveOneOut()
loo.get_n_splits(x)
# x_train mane whole data nea dorkar ekhane
for train_index, test_index in loo.split(x):
        #print("train:", train_index, "validation:", test_index)
        x_train, x_test = x[train_index], x[test_index]
        y_train, y_test = y[train_index], y[test_index]
        
        
        model.fit(x_train, y_train.ravel())
        y_train_pred =  model.predict(x_train)  #--------------------------------------------------
        #print (y_train_true.values.ravel(),y_train_pred)

        #y_test_pred = model.predict(x_test)        
#model.fit(x_train, y_train_true.values.ravel()) ###########################################

print('Best C:',model.best_estimator_.C) 
print('Best Kernel:',model.best_estimator_.kernel)
print('Best Gamma:',model.best_estimator_.gamma)

from sklearn import metrics
from sklearn.metrics import confusion_matrix 
from sklearn.metrics import accuracy_score 
from sklearn.metrics import classification_report 

print("Train Accuracy:",metrics.accuracy_score(y_train, y_train_pred))

results = confusion_matrix(y_train, y_train_pred)
print ('Confusion Matrix :')
print(results) 
print ('Accuracy Score :',accuracy_score(y_train, y_train_pred) ) 
print ('Report : ')
print (classification_report(y_train, y_train_pred) )

Train Accuracy: 0.610687022901
Confusion Matrix :
[[38  3  8]
 [16  9 11]
 [ 8  5 33]]
Accuracy Score : 0.610687022901
Report : 
             precision    recall  f1-score   support

          1       0.61      0.78      0.68        49
          2       0.53      0.25      0.34        36
          3       0.63      0.72      0.67        46

avg / total       0.60      0.61      0.59       131



# Same model 90% random data

In [0]:
parameter_candidates = [
  {'C': [1, 10, 100, 1000], 'kernel': ['linear']},
  {'C': [1, 10, 100, 1000], 'gamma': [0.001, 0.0001], 'kernel': ['rbf']},
]
# Create a classifier object with the classifier and parameter candidates
model = GridSearchCV(estimator=svm.SVC(), param_grid=parameter_candidates, n_jobs=-1)



from sklearn.model_selection import LeaveOneOut ########LEAVE ONE OUT
import numpy as np
#x = np.array([[1, 2], [3, 4]])
#x_train = data_avn_fan_op.iloc[:,[0]] #----------------
#y_train = data_avn_fan_op.iloc[:,[0]]


x = np.array(x_train)
y = np.array(y_train.values.ravel())
loo = LeaveOneOut()
loo.get_n_splits(x)
# x_train mane whole data nea dorkar ekhane
for train_index, test_index in loo.split(x):
        #print("train:", train_index, "validation:", test_index)
        x_train, x_test = x[train_index], x[test_index]
        y_train, y_test = y[train_index], y[test_index]
        
        
        model.fit(x_train, y_train.ravel())
        y_train_pred =  model.predict(x_train)  #--------------------------------------------------
        #print (y_train_true.values.ravel(),y_train_pred)

        #y_test_pred = model.predict(x_test)

        
#model.fit(x_train, y_train_true.values.ravel()) ###########################################

#mean absolute value for training data
##data = train[target]

#print(y_test_true.values.ravel(), y_test_pred)


from sklearn import metrics

print("Train Accuracy:",metrics.accuracy_score(y_train, y_train_pred))


from sklearn.metrics import confusion_matrix 
from sklearn.metrics import accuracy_score 
from sklearn.metrics import classification_report 

results = confusion_matrix(y_train, y_train_pred)
print ('Confusion Matrix :')
print(results) 
print ('Accuracy Score :',accuracy_score(y_train, y_train_pred) ) 
print ('Report : ')
print (classification_report(y_train, y_train_pred) )

Train Accuracy: 1.0
Confusion Matrix :
[[49  0  0]
 [ 0 36  0]
 [ 0  0 46]]
Accuracy Score : 1.0
Report : 
             precision    recall  f1-score   support

          1       1.00      1.00      1.00        49
          2       1.00      1.00      1.00        36
          3       1.00      1.00      1.00        46

avg / total       1.00      1.00      1.00       131



# Random forest ----leave one out & AVN_FAN

In [0]:
#Import Random Forest Model
from sklearn.ensemble import RandomForestClassifier

clf=RandomForestClassifier(n_estimators=400)
model = clf

#clf.fit(x_train,y_train_true)
#y_test_pred=clf.predict(x_test)
#x_train = data_avn_fan.iloc[:,1:17]
x_train = data_avn_fan_op.iloc[:,1:22] #----------------
y_train = data_avn_fan_op.iloc[:,[0]]


x = np.array(x_train)
y = np.array(y_train.values.ravel())
loo = LeaveOneOut()
loo.get_n_splits(x)
for train_index, test_index in loo.split(x):
        #print("train:", train_index, "validation:", test_index)
        x_train, x_test = x[train_index], x[test_index]
        y_train, y_test = y[train_index], y[test_index]
        
        
        model.fit(x_train, y_train.ravel())
        y_train_pred =  model.predict(x_train)  #--------------------------------------------------
        #print (y_train_true.values.ravel(),y_train_pred)

        #y_test_pred = model.predict(x_test)        
#model.fit(x_train, y_train_true.values.ravel()) ###########################################

from sklearn import metrics
from sklearn.metrics import confusion_matrix 
from sklearn.metrics import accuracy_score 
from sklearn.metrics import classification_report 

print("Train Accuracy:",metrics.accuracy_score(y_train, y_train_pred))

results = confusion_matrix(y_train, y_train_pred)
print ('Confusion Matrix :')
print(results) 
print ('Accuracy Score :',accuracy_score(y_train, y_train_pred) ) 
print ('Report : ')
print (classification_report(y_train, y_train_pred) )

Train Accuracy: 0.908396946565
Confusion Matrix :
[[47  1  1]
 [ 4 31  1]
 [ 5  0 41]]
Accuracy Score : 0.908396946565
Report : 
             precision    recall  f1-score   support

          1       0.84      0.96      0.90        49
          2       0.97      0.86      0.91        36
          3       0.95      0.89      0.92        46

avg / total       0.91      0.91      0.91       131



AVN & FAN - Random forest-400
Train Accuracy: 0.862595419847
Confusion Matrix :
[[45  0  4]
 [ 7 28  1]
 [ 4  2 40]]
Accuracy Score : 0.862595419847
Report : 
             precision    recall  f1-score   support

          1       0.80      0.92      0.86        49
          2       0.93      0.78      0.85        36
          3       0.89      0.87      0.88        46

avg / total       0.87      0.86      0.86       131

AVN & FAN & OP- Random forest-400
Accuracy: 0.908396946565
Confusion Matrix :
[[47  1  1]
 [ 4 31  1]
 [ 5  0 41]]
Accuracy Score : 0.908396946565
Report : 
             precision    recall  f1-score   support

          1       0.84      0.96      0.90        49
          2       0.97      0.86      0.91        36
          3       0.95      0.89      0.92        46

avg / total       0.91      0.91      0.91       131

# ERROR for REGression

In [0]:



test_data_error = mean_absolute_error(y_test_true, y_test_pred)
accuracy

In [0]:
#define model I am using
model = LinearRegression()
#training process
##model.fit(train[features], train[target])
from sklearn.metrics import f1_score

model.fit(data_avn, target_1col)
#mean absolute value for training data
##data = train[target]
true_target = target_1col
##predict =  model.predict(train[features])
predict =  model.predict(data_avn)


training_error = mean_absolute_error(true_target, predict)
accuracy = r2_score(true_target, predict)
#accuracy = f1_score(true_target, predict, average='micro')
print(predict)
accuracy




[[  2799.4262873 ]
 [  2833.22335692]
 [  2492.58545066]
 [  2551.87960068]
 [  2000.29881734]
 [  2124.05002439]
 [  2365.65288887]
 [  2001.82331314]
 [  5050.11742108]
 [  1873.36625554]
 [  1873.36625554]
 [  1873.36625554]
 [  1873.36625554]
 [ 13033.93275207]
 [  2688.02791551]
 [  3862.24907869]
 [  2365.65288887]
 [  2336.39627142]
 [  2773.44160075]
 [  1873.36625554]
 [  1873.36625554]
 [  1526.43521463]
 [  1286.09171366]
 [  2034.09588696]
 [  7713.61507356]
 [  1873.36625554]
 [  3045.56960396]
 [  1873.36625554]
 [  1873.36625554]
 [  1873.36625554]
 [  1873.36625554]
 [  1032.22659007]
 [  6963.05424183]
 [  2619.51801246]
 [  1873.36625554]
 [  2709.47214988]
 [  3975.29906205]
 [  1873.36625554]
 [  2463.32883322]
 [  2828.68290475]
 [  2336.39627142]
 [  2799.4262873 ]
 [  5239.91840605]
 [  1661.02000851]
 [  2364.06560477]
 [  1873.36625554]
 [  3484.2580694 ]
 [  2000.29881734]
 [  3987.65714468]
 [  3387.98549501]
 [  1873.36625554]
 [  1873.36625554]
 [  7222.134

0.39671025891021217

In [0]:
#define model I am using
model = LinearRegression()
#training process
##model.fit(train[features], train[target])


model.fit(data_avn_fan, target_1col)
#mean absolute value for training data
##data = train[target]
true_target = target_1col
##predict =  model.predict(train[features])
predict =  model.predict(data_avn_fan)


training_error = mean_absolute_error(true_target, predict)
accuracy2 = r2_score(true_target, predict)

accuracy2

0.46434284014780314

In [0]:
#mean absolute value for test data

##test_data = test[target]
test_data = test.iloc[:,[1]]
test_data = test['wml']
##predict_test = model.predict(test[features])
test_data_avn=test.iloc[:,[12,13,14,15,16,17,30,31,32,33,34,35]]
test_data_fan=test.iloc[:,[18,19,20,21,22,23, 36,37,38,39,40,41]]

predict_test = model.predict(test_data_avn)
test_data_error = mean_absolute_error(test_data, predict_test)
accuracy

ValueError: shapes (14,12) and (24,1) not aligned: 12 (dim 1) != 24 (dim 0)

In [0]:
#we need some metric to measure the accuracy of our regression model
from sklearn.metrics import r2_score
#on training data
true_value = data
predicted_val =  model.predict(data_avn)
accuracy = r2_score(true_value, predicted_val)
print(accuracy)
#on test data
true_value2 = test_data
predicted_val2 =  model.predict(test_data_avn)
accuracy3 = r2_score(true_value2, predicted_val2)
print(accuracy2)

0.128221131561
0.130647591078
